plugins/gradle/java/src/execution/test/runner/AllInPackageGradleConfigurationProducer.java

```diff
@@ -55,6 +60,12 @@ protected boolean doSetupConfigurationFromContext(ExternalSystemRunConfiguration
     String filter = createTestFilterFrom(configurationData.psiPackage, /*hasSuffix=*/false);
     configuration.getSettings().setScriptParameters(filter);
     configuration.setName(suggestName(configurationData));
+    // Store information about the location of the run configuration.
+    Module configurationModule = ((ModuleWithDependenciesScope)configuration.getSearchScope()).getModule();
+    VirtualFile locationFile = getSourceFile(getSourceElement(configurationModule, configurationData.sourceElement));
+    if (locationFile != null) {
+      configuration.putUserData(LOCATION_NAME, locationFile);
```

Comment:
This property is not shown in UI, so there are at least 2 scenarios that would look buggy to a user, and original problem (unit tests instead of android tests are executed) will arise:
1. User creates run configuration manually
2. User copies run configuration (via "copy" button)